dask/array/blockwise.py

```diff
@@ -6,7 +6,62 @@
 from .. import base, utils
 from ..delayed import unpack_collections
 from ..highlevelgraph import HighLevelGraph
-from ..blockwise import blockwise as core_blockwise
+from ..blockwise import (
+    Blockwise,
+    blockwise as core_blockwise,
+    blockwise_token,
+)
+
+
+class BlockwiseCreateArray(Blockwise):
+    """
+    Specialized Blockwise Layer for array creation routines.
+
+    Enables HighLevelGraph optimizations.
+
+    Parameters
+    ----------
+    name: string
+        The output name.
+    func : callable
+        Function to apply to populate individual blocks. This function should take
+        an iterable containing the dimensions of the given block.
+    shape: iterable
+        Iterable containing the overall shape of the array.
+    chunks: iterable
+        Iterable containing the chunk sizes along each dimension of array.
+    """
+
+    def __init__(
+        self,
+        name,
+        func,
+        shape,
+        chunks,
+    ):
+        # self.name = name
+        io_name = "blockwise-create-" + name
+
+        # Define "blockwise" graph
+        dsk = {name: (func, blockwise_token(0))}
+
+        out_ind = tuple(range(len(shape)))
+        self.nchunks = tuple(len(chunk) for chunk in chunks)
+        super().__init__(
+            name,
+            out_ind,
+            dsk,
+            [(io_name, out_ind)],
+            {io_name: self.nchunks},
+            io_deps={
+                io_name: {
+                    "func": (
+                        "generate_array_chunk",
+                        chunks,
+                    )
+                }
+            },
```

Comment:
@ian-r-rose - Here is one possible approach to the dynamic chunk-generation problem we discussed in #7042 

The idea is to specify a "key" (`"generate_array_chunk"`) for a registered function that is already known to the scheduler.  This seems to work fine, but I am very open to more-elegant solutions :)